cli/src/config.rs

```diff
@@ -0,0 +1,121 @@
+use color_eyre::Result;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::{fs::File, io::BufReader, net::SocketAddr, path::PathBuf, time::Duration};
+use tower_http::cors::CorsLayer;
+
+/// Server Configuration passed on initialization
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    /// Server address <host:port>
+    pub address: SocketAddr,
+
+    /// Enable Default Cors configuration
+    #[serde(default = "default_cors")]
+    pub cors_allow_all: bool,
+
+    /// Max Time to Responce, in milliseconds
+    #[serde(default = "default_timeout")]
+    #[serde(with = "humantime_serde")]
+    pub request_timeout: Duration,
+
+    /// [`Logger`](LoggerConfig) Configuration
+    #[serde(flatten)]
+    pub logger: LoggerConfig,
+}
+
+/// Logger Configuration passed on initialization
+#[allow(clippy::module_name_repetitions)]
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct LoggerConfig {
+    /// [Stub] File to save logs
+    pub log_file: Option<PathBuf>,
+
+    /// [Stub] Number of log files
+    #[serde(default = "default_log_amount")]
+    pub log_amount: usize,
+
+    /// [Stub] Max size of a single log file, in bytes
+    #[serde(default = "default_log_size")]
+    pub log_size: u64,
+
+    /// Tracing Level
+    #[serde(default = "tracing_default")]
+    #[serde_as(as = "DisplayFromStr")]
+    pub trace_level: tracing::Level,
+}
+
+impl Config {
+    /// Return either very permissive [`CORS`](`CorsLayer`) configuration
+    /// or empty one based on `cors_allow_all` field
+    pub fn get_cors_configuration(&self) -> CorsLayer {
```

Comment:
It is strange to see `CorsLayer` dependency on Cli level. `Cors` are specific to implementation, therefore this conversion should be inside the backend code